torch/onnx/_patch_torch.py

```diff
@@ -129,7 +129,7 @@ def _new_node(
     Returns:
         The new node.
     """
-    aten = kwargs.pop("aten", False)
+    aten = kwargs.pop("aten", False) or (namespace == "aten" and op == "ATen")
```

Comment:
I think we can get rid of the `aten` argument (aka `_add_attribute(..., aten)`  and `kwargs["aten"]` entirely and rely only on namespace/opname for identifying aten op fallback

Make sure you build pytorch with caffe2 and check whether https://github.com/pytorch/pytorch/blob/master/caffe2/contrib/aten/aten_test.py passes. There are some test/onnx/test_*caffe2*.py that we should cofnirm are working too. Not sure whether public CI still call them or only from time to time on internal Meta CIs

Reply:
looks like aten is True even when op is not ATen: seen in block op

```
def _block_op(b: _C.Block, opname: str, *args, **kwargs):
    if "::" in opname:
        aten = False
        ns_opname = opname
    else:
        aten = kwargs.pop("aten", False)
        ns = "aten" if aten else "onnx"
        ns_opname = ns + "::" + opname
    n = b.addNode(ns_opname, args)
    for k, v in sorted(kwargs.items()):
        if k == "inplace":
            continue
        _add_attribute(n, k, v, aten=aten)
    outputs = tuple(n.outputs())
    if len(outputs) == 1:
        return n.output()
    return outputs
```

Reply:
Indeed, `_block_op` assumes aten operators will not have domain set, because empty namespace means `aten` already. However, https://github.com/pytorch/pytorch/pull/83136/  breaks this contract and add `aten::ATen` instead of `ATen`.

Check if this assumption is broken in other places too

Reply:
It doesn't seem like _add_attribute takes the namespace information we can use?

Reply:
Updated contract for _block_op. I haven't found code in our code base that calls for aten ops in a block.

Reply:
empty namespace typically mean `onnx`, unless `aten=True` is passed in as a keyword arg. Do you think we should (1) assume `aten=True` can be set by users and retain the logic for handling it or (2) Remove the logic for handing `aten=True` since we are not using it in pytorch anyways. I expect (2) will impact `block.op(aten=True)` calls more than graph.op calls, since users typically call `graph.at` to create aten ops, which we kept the interface stable.